consensus/replay.go

```diff
@@ -417,6 +417,12 @@ func (h *Handshaker) replayBlocks(state sm.State, proxyApp proxy.AppConns, appBl
 	for i := appBlockHeight + 1; i <= finalBlock; i++ {
 		h.logger.Info("Applying block", "height", i)
 		block := h.store.LoadBlock(i)
+
+		// Extra check to ensure the app was not changed in a way it shouldn't have.
+		if len(appHash) > 0 {
+			checkAppHashAgainstBlock(appHash, block)
```

Comment:
Why is the error never checked here? And why does the method never return an error (only panic).